code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -567,85 +501,67 @@
 	model_type = list(/obj/item/robot_model/medical, /obj/item/robot_model/syndicate_medical)
 	model_flags = BORG_MODEL_MEDICAL
 
-/obj/item/borg/upgrade/processor/action(mob/living/silicon/robot/R, user = usr)
-	. = ..()
-	if(.)
-		var/obj/item/borg/upgrade/processor/U = locate() in R
-		if(U)
-			to_chat(user, span_warning("This unit is already equipped with a surgical processor module!")) //check to see if we already have this module
-			return FALSE
-		var/obj/item/surgical_processor/SP = new(R.model)
-		R.model.basic_modules += SP
-		R.model.add_module(SP, FALSE, TRUE)
-
-/obj/item/borg/upgrade/processor/deactivate(mob/living/silicon/robot/R, user = usr)
-	. = ..()
-	if (.)
-		var/obj/item/surgical_processor/SP = locate() in R.model
-		R.model.remove_module(SP, TRUE)
+	items_to_add = list(/obj/item/surgical_processor)
 
 /obj/item/borg/upgrade/ai
-	name = "B.O.R.I.S. module"
+	name = "B.O.borg.I.S. module"
 	desc = "Bluespace Optimized Remote Intelligence Synchronization. An uplink device which takes the place of an MMI in cyborg endoskeletons, creating a robotic shell controlled by an AI."
 	icon_state = "boris"
 
-/obj/item/borg/upgrade/ai/action(mob/living/silicon/robot/R, user = usr)
+/obj/item/borg/upgrade/ai/action(mob/living/silicon/robot/borg, user = usr)
 	. = ..()
 	if(.)
-		if(locate(/obj/item/borg/upgrade/ai) in R.upgrades)
-			to_chat(user, span_warning("This unit is already an AI shell!"))
-			return FALSE
-		if(R.key) //You cannot replace a player unless the key is completely removed.
-			to_chat(user, span_warning("Intelligence patterns detected in this [R.braintype]. Aborting."))
+		if(borg.key) //You cannot replace a player unless the key is completely removed.
+			to_chat(user, span_warning("Intelligence patterns detected in this [borg.braintype]. Aborting."))
 			return FALSE
 
-		R.make_shell(src)
+		borg.make_shell(src)
 
-/obj/item/borg/upgrade/ai/deactivate(mob/living/silicon/robot/R, user = usr)
+/obj/item/borg/upgrade/ai/deactivate(mob/living/silicon/robot/borg, user = usr)
 	. = ..()
 	if (.)
-		if(R.shell)
-			R.undeploy()
-			R.notify_ai(AI_NOTIFICATION_AI_SHELL)
+		if(borg.shell)
+			borg.undeploy()
+			borg.notify_ai(AI_NOTIFICATION_AI_SHELL)
```

Comment:
yeah look at how this is right now and compare it to how nice it looks if we did

```dm
if(!. || !borg.shell) // or isnull here if shell is a typed variable
 return .

borg.undeploy()
borg.notify_ai(AI_NOTIFICATION_AI_SHELL)
```

looks cleaner to me